### PR TITLE
[MRG] modify signature filtering to avoid identical md5sum issues with `sig grep`, `sig extract`

### LINKED ISSUE #2593: unexpected `sig grep`, `sig extract` behavior with identical md5s
`sig grep` produces an unintuitive result if sigs have identical md5's:

```
sourmash sig grep "AY770193" output.spillover/spillover100.dna.zip -k 21 --dna > test.sig

== This is sourmash version 4.8.2. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

saving matching signatures to '-'
loaded 100 total that matched ksize & molecule type
extracted 2 signatures from 1 file(s)
```
And the signatures:
```
sourmash sig describe test.sig

== This is sourmash version 4.8.2. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

---
signature filename: test.sig
signature: AY770193 Andes orthohantavirus
source file: output.spillover/genomic/AY770193.fna.gz
md5: 31d1b130702bc77a42d3876259fe2835
k=21 molecule=DNA num=0 scaled=1 seed=42 track_abundance=1
size: 212
sum hashes: 212
signature license: CC0

---
signature filename: test.sig
signature: AY770195 Andes orthohantavirus
source file: output.spillover/genomic/AY770195.fna.gz
md5: 31d1b130702bc77a42d3876259fe2835
k=21 molecule=DNA num=0 scaled=1 seed=42 track_abundance=1
size: 212
sum hashes: 212
signature license: CC0

loaded 2 signatures total, from 1 files
```

if I grep for "AY770195" , I get the same two signatures.
But, if I grep for "AY77019" instead, I get 8 signatures, all of the AY77019*.

As Mo pointed out, the md5sums are identical. I think sig grep is using the md5 to pull out the signatures, which is why I'm getting both of these signatures here, despite their name differences. 

Note, the same thing happens with `sig extract` (which makes sense, because I think it uses a similar strategy).

```
sourmash sig extract -k 21 --dna --name  "AY770193 Andes orthohantavirus" output.spillover/spillover100.dna.zip > test6.sig
```
```
== This is sourmash version 4.8.2. ==
== Please cite Brown and Irber (2016), doi:10.21105/joss.00027. ==

loaded 100 total that matched ksize & molecule type
extracted 2 signatures from 1 file(s)
```

I was testing some extremely short sequences here (and the reference sequences themselves seem to be identical), so running into identical md5's not super surprising, though I'll look into it. But we should still consider how to handle behavior in these cases, as identical md5's do happen (albeit more rarely) with our more standard uses.

This db can be found on farm: `/home/ntpierce/2023-spillover/output.spillover/spillover100.dna.zip`.

One of my main issues with this behavior is that if you build a single zip database, there's no way to extract these signatures separately for downstream steps, e.g. `gather` against a reference database.

### PR DESCRIPTION
**Pull request by bluegenes**
Fixes https://github.com/sourmash-bio/sourmash/issues/2593

Problem: when two signatures with different signature metadata (name, etc) but the same md5sum are in a single index, grepping/extracting via metadata (e.g. with a name) will return all signatures with matching md5sum.

Here, I add capacity to further filter signatures using the sub_manifest created during `sig grep`/`sig extract`, so that we only return the signatures with correct metadata.

### CODE REVIEW

**Comment by bluegenes** [comment]
@ctb @sourmash-bio/devs ready for review!

File: src/sourmash/sig/grep.py [hunk 1]
```diff
@@ -97,9 +97,8 @@ def main(args):
         elif not args.silent:
             # nope - do output signatures. convert manifest to picklist, apply.
             sub_picklist = sub_manifest.to_picklist()
-
             try:
-                idx = idx.select(picklist=sub_picklist)
+                idx = idx.select(picklist=sub_picklist, filter_manifest=sub_manifest)
```
Comment (bluegenes) [inline src/sourmash/sig/grep.py:101]: @ctb I thought I might be able to get rid of using the `picklist=sub_picklist` here, but I think it's needed for LCA databases and databases with no manifests.

**Comment by ctb** [comment]
I'm planning to review this over the weekend!

**Comment by ctb** [comment]
Fixed instead by #2747

**Pull request closed without merging**